eventlog/tcg2_test.go

```diff
@@ -49,16 +49,15 @@ var (
 	goodStruct = &SP800155Event3{
 		PlatformManufacturerID:  11129,
 		ReferenceManifestGUID:   EfiGUID{UUID: uuid.MustParse(rimGUID)},
-		PlatformManufacturerStr: ByteSizedArray{Data: []byte("Google Compute Engine")},
-		PlatformModel:           ByteSizedArray{Data: []byte("PFModel")},
-		PlatformVersion:         ByteSizedArray{Data: []byte("PFVersion")},
-		FirmwareManufacturerStr: ByteSizedArray{Data: []byte("Vanadium")},
+		PlatformManufacturerStr: ByteSizedCStr{Data: "Google Compute Engine"},
+		PlatformModel:           ByteSizedCStr{Data: "PFModel"},
+		PlatformVersion:         ByteSizedCStr{Data: "PFVersion"},
+		FirmwareManufacturerStr: ByteSizedCStr{Data: "Vanadium"},
 		FirmwareManufacturerID:  54494,
-		FirmwareVersion:         ByteSizedArray{Data: []byte("2.0.0")},
+		FirmwareVersion:         ByteSizedCStr{Data: "2.0.0"},
 		RIMLocatorType:          RIMLocationVariable,
 		RIMLocator:              Uint32SizedArray{Data: append(myEfiGUID, 'V', 0, 'a', 0, 'r', 0, 0, 0)},
 		PlatformCertLocatorType: RIMLocationRaw,
-		PlatformCertLocator:     Uint32SizedArray{Data: []byte{}},
```

Comment:
PlatformCertLocator isn't needed anymore? Or is the default good here?

Reply:
The default is good. That's the logic I added to make size == 0 => data = nil instead of []byte{}.